CONTRIBUTING.md

```diff
@@ -0,0 +1,37 @@
+# Contributing
+
+Contributions are welcome!
+
+## Adding a new distro
+
+### Add Dockerfiles
+
+Create `build_<DISTRO>.dockerfile` and `run_<DISTRO>.dockerfile` in the root directory.
+
+* Add one [stage](https://docs.docker.com/develop/develop-images/multistage-build/)
+  for each (supported) backend: `mcode`, `llvm` and/or `gcc`.
+* Images should contain the dependencies for building or installing GHDL only.
+  Do not include the actual building/installation procedures in the dockerfiles.
+
+### Continuous Integration (CI)
+
+* In `run.sh`, add a new `case` inside of `create ()`.
+* Add the desidered `<DISTRO>` and its `<VERSIONS>` to the matrix of the `base.yml` CI workflow:
+```yaml
+matrix:
+  include: [
+    { distro: arch,     version: ''        },
+    { distro: debian,   version: buster    },
+    ...
+    { distro: <DISTRO>, version: <VERSION> },
+```
+* Add the desidered `<DISTRO><VERSION>` and `<BACKEND>` to the matrix of the `test.yml` CI workflow:
```

Comment:
Maybe we should clarify that the version of the backend needs to be specified for GCC and for LLVM on Debian/Ubuntu: https://github.com/ghdl/docker/blob/master/.github/workflows/test.yml#L20-L37

Reply:
Sorry, I have not clear that point. I don't know why is specified with Debian and Ubuntu, and not in Fedora, and in the case of centos, is not the version of the backend, is the version of the OS (something to provide a different stage name).

Reply:
On Fedora, the argument to `configure` is `--with-llvm-config=llvm-config`. However, on Debian/Ubuntu executable `llvm-config` does not exist. It is named `llvm-config-VERSION`. That's why we need to specify it.

On the other hand, on CentOS, the version of the distribution is provided in field `distro`. Field `backend` contains the name and version of the backend. Hence `distro: centos7,  backend: gcc-7` is read as "build GCC v7 on CentOS 7". The same way `distro: bullseye, backend: gcc-9.1.0` means "build GCC v9.1.0 on Debian Bullseye". It's unfortunate that you chose to use GCC v7 on CentOS 7 and GCC v8 on CentOS 8. That's misleading.

>  is the version of the OS (something to provide a different stage name)

This is the case in workflow `base.yml`, NOT in `test.yml` (which we are talking about here).

Reply:
Hi. I tried to clarify this and other points. Let me know what do you think.